src/File/Csv.php

```diff
@@ -25,9 +27,40 @@ public function convert(): string
     {
         $data = $this->parseData();
         $keys = $this->parseCsv(array_shift($data));
-        return json_encode(array_map(function ($line) use ($keys) {
-            return array_combine($keys, $this->parseCsv($line));
-        }, $data), $this->conversion['options']);
+        $splitKeys = array_map(function ($key) {
+            return explode($this->conversion['join'], $key);
+        }, $keys);
+
+        $jsonObjects = array_map(function ($line) use ($splitKeys) {
+            $values = $this->parseCsv($line);
+            $jsonObject = [];
+            for ($valueIndex = 0; $valueIndex < count($values); $valueIndex++) {
```

Comment:
Since for loop evaluates the second expression every time in each iteration, I think it'd be better with something below.

```php
for ($valueIndex = 0, $count = count($values); $valueIndex < $count; $valueIndex++) {
```

Reply:
Well spotted. I've updated it